packages/vite/src/node/server/moduleGraph.ts

```diff
@@ -143,6 +155,10 @@ export class ModuleGraph {
       // processing being done for this module
       mod.lastInvalidationTimestamp = timestamp
     }
+    if (softInvalidate) {
+      mod.softInvalidatedTransformResult = mod.transformResult
```

Comment:
I think you broke the softDelete -> hardDelete because in that case softInvalidatedTransformResult will still be non-null and be served on the next request?

Reply:
Argh, you're right. Yeah the whole thing is still in flux because since the first commit I realized:

1. We need to have separate soft-invalidations for client and ssr, since both can be requested at different timings.
2. If a module (A) `addWatchFile` another module (B), the relation is `A` imports `B` in the module graph. If `B` changes, `A` cannot be soft-invalidated as it needs to retransforms itself.

I'm still in the midst of tackling both, so most of the code isn't quite ready for review yet 😅

Reply:
Ok re-request a review when ready !